### PR TITLE
Feature/tag list

### PR DESCRIPTION
**Pull request by rushton3179**

### CODE REVIEW

File: xbdd-web/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import TagListContainer from "./modules/tag-list/containers/TagListContainer";
+import Report from "./models/Report";
+import "./App.css";
+import tagListData from "./resources/tag-response.json";
+
+const App = () => {
+    const dummyReport = new Report(tagListData);
```
Comment (yashitkumarsingh) [inline xbdd-web/src/App.jsx]: We should add TODOs for the stuff we want to remove later?

File: xbdd-web/src/__tests__/tag-list/components/TagListFilterButtonsView-test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import TagListFilterButtonsView from "../../../modules/tag-list/components/TagListFilterButtonsView";
+
+const filterStates = {
+    passedSelected: true,
+    undefinedSelected: true,
+    failedSelected: true,
+    skippedSelected: true,
+};
+
+test("TagListButtonsView renders", () => {
```
Comment (yashitkumarsingh) [inline xbdd-web/src/__tests__/tag-list/components/TagListFilterButtonsView-test.js]: Probably best to wrap it in a describe block having 'TagListButtonsView'  and then it blocks with 'renders ....' ?

File: xbdd-web/src/__tests__/tag-list/components/TagListItemView-test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { shallow, mount } from "enzyme";
+import TagListItemView from "../../../modules/tag-list/components/TagListItemView";
+import Tag from "../../../models/Tag";
+
+const dummyTag = new Tag({
+    tag: "testTagOne",
+    features: [],
+});
+
+test("TagListItemView renders", () => {
```
Comment (yashitkumarsingh) [inline xbdd-web/src/__tests__/tag-list/components/TagListItemView-test.js]: Wrap it in a describe block?

Comment (rushton3179) [reply xbdd-web/src/__tests__/tag-list/components/TagListItemView-test.js]: Done

File: xbdd-web/src/__tests__/tag-list/components/TagListFilterButtonsView-test.js (hunk 2 above)
Comment (rushton3179) [reply xbdd-web/src/__tests__/tag-list/components/TagListFilterButtonsView-test.js]: Done

File: xbdd-web/src/App.jsx (hunk 1 above)
Comment (rushton3179) [reply xbdd-web/src/App.jsx]: kk

**Pull request merged**